### DISCUSSION TITLE
is there a way to have vscode to install node_modules in the extension folder

### DISCUSSION THREAD

**Discussion by c3wenjiaowang** [microsoft/vscode-discussions#190] [Extension Development QnA] [answered]
Found if we don't manually trigger a `npm install` at the extension folder (under ~/.vscode/extension/<extension name>), vscode is not doing an npm install to download all dependencies into `node_modules` folder. Is there a way that we can configure so that vscode does that for us? As when we post the extension onto market place, there is 25 MB size limit, and that makes us not able to pack the `node_modules` into the package, we have to run npm install in the extension folder after installing the vsix file to download the dependencies. There is a similar post on this issue, and seems like there was not a solution at that time: https://github.com/microsoft/vscode-vsce/issues/100, and seems it is a universal issue that would be better handled by vs-code.

**Comment by r3m0t** [comment, marked as answer]
No, the vsix has to contain everything for your extension to start, without extra downloads.

If you use webpack or esbuild then it can include only the relevant `.js` files from `node_modules` and also minify them to reduce the file size. Then you can add `node_modules` to the `.vscodeignore` . For example, `node_modules` folder here is 600 MB - https://github.com/microsoft/vscode-jupyter - but the download size is 15 MB.

Have a look here - 
https://code.visualstudio.com/api/working-with-extensions/bundling-extension
